### DISCUSSION TITLE
How to Access .JSON and .p4 Files After Compiling a P4 Program in Kathara?

### DISCUSSION THREAD

**Discussion by em-saqib** [Kidel/Kathara#304] [Q&A]
For example, if we write p4c program.p4 in the startup file of the switch, where does the compiler store the resulting JSON and .p4 files after the compilation process?

Thanks.

**Comment by Skazza94** [comment]
Hi @em-saqib,
the compiled files for the P4 program are, by default, stored in the same folder of the `.p4`.

In case you are using the Kathara-Labs, we place the `.p4` files in the root of the container (the `/` folder). 
So, from a device terminal window, if you run:
```bash
ls -la /
```
You will be able to see the `.json` of the compiled P4 program.

You can also change the folder by using the `-o` parameter of the `p4c` command, for example:
```bash
p4c -o /my/custom/path program.p4
```

**Reply by em-saqib** [reply]
Hi Mariano,

When I type **ls -la /**, I see some root dirs but don't see the P4 compiled files.
For the 2nd option to store the compiled files to my desired directory, it generate the fies when I set shared folder. For example: **p4c -o /sahred program.p4**

But the switch cannot connect and log file says: **the info/json file cannot be loaded.**

**Reply by Skazza94** [reply]
Hi @em-saqib,
it's a bit difficult to understand what happens without providing any screenshot/file. 

Can you upload a zip of the scenario so that I can check it? 

Thanks.
